slm_porosity/heat_transfer.f90

```diff
@@ -755,26 +763,43 @@ SUBROUTINE user_pre_process
     Dphi = liquid_fraction(h, 1)
     Dpsi = porosity(u(:,n_var_porosity), phi, Dphi)
 
+    ! NB: 1) ng is not equal to nwlt after mesh adaptation
+    !     2) ne is not equal to n_integrated until first call of time integration method
+    for_du = RESHAPE((/ h_star, temp /), SHAPE(for_du))
+    CALL c_diff_fast(for_du, du, du_dummy, j_lev, nwlt, grad_meth(meth), 10, nvar, 1, nvar)
+
+    ! 2. Store the calculated variables as global variables with suffix "_prev"
     ! quantities used in DRHS
     CALL reallocate_scalar(diffusivity_prev); diffusivity_prev = diffusivity(h, psi)
     CALL reallocate_scalar(Ddiffusivity_prev); Ddiffusivity_prev = diffusivity(h, psi, Dpsi)
     CALL reallocate_scalar(Dh_star_prev); Dh_star_prev = enthalpy_star(h, 1)
+    CALL reallocate_vector(grad_h_star_prev); grad_h_star_prev = du(1,:,:)
 
     ! quantities used in algebraic BC
     CALL reallocate_scalar(enthalpy_prev); enthalpy_prev = h
     CALL reallocate_scalar(temp_prev); temp_prev = temp
     CALL reallocate_scalar(Dtemp_prev); Dtemp_prev = temperature(h, temp)
     CALL reallocate_scalar(psi_prev); psi_prev = psi
 
-    ! NB: 1) ng is not equal to nwlt after mesh adaptation
-    !     2) ne is not equal to n_integrated until first call of time integration method
-    for_du = RESHAPE((/ h_star /), SHAPE(for_du))
-    CALL c_diff_fast(for_du, du, du_dummy, j_lev, nwlt, grad_meth(meth), 10, n_integrated, 1, n_integrated)
-    CALL reallocate_vector(grad_h_star_prev); grad_h_star_prev = du(1,:,:)
+
+    ! 3. Check if J_MX provides a sufficient resolution for solid--liquid interface thickness
+    IF (check_resolution) THEN
+      max_grad_temp = MAXVAL(ABS(du(2,:,:)), 1)
+      CALL parallel_vector_sum(REALMAXVAL=max_grad_temp, LENGTH=dim)
+      thickness = interface_thickness/max_grad_temp
+      resolution = (xyzlimits(2,:) - xyzlimits(1,:)) / (mxyz*2**j_mx)
```

Comment:
Does this formula have any physical meaning?

Reply:
`thickness` has a physical meaning. We want to resolve it and, therefore, to compare with the minimum mesh step.